java/refactoring.java/src/org/netbeans/modules/refactoring/java/ui/CopyFullyQualifiedClassNameAction.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.netbeans.modules.refactoring.java.ui;
+
+import java.awt.Toolkit;
+import java.awt.datatransfer.StringSelection;
+import org.netbeans.api.java.classpath.ClassPath;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionReferences;
+import org.openide.awt.ActionRegistration;
+import org.openide.filesystems.FileObject;
+import org.openide.nodes.Node;
+import org.openide.util.HelpCtx;
+import org.openide.util.NbBundle;
+import org.openide.util.actions.NodeAction;
+
+/**
+ * Action to copy fully qualified name of selected classes.
+ * 
+ * @author Benjamin Asbach
+ */
+@ActionID(id = "org.netbeans.modules.refactoring.java.java.ui.CopyFullyQualifiedClassNameAction", category = "Edit")
+@ActionRegistration(displayName = "Copy Fully Qualified Class Name", lazy = false)
```

Comment:
Is the lazy=false needed here? Isn't there a way to add a non-lazy action that is enabled only when a FileObject is in the lookup?

Reply:
I'm unsure I looked at some other actions. I'm not 100% certain what's all happening with `lazy` `true` vs `false`. Changed it to `true` and tested it which worked.